### ISSUE TITLE
Nullable array keys are allowed in type annotations

### ISSUE THREAD

**Issue by ssandler** [facebook/hhvm#8628]
### HHVM Version

```
$ hhvm --version
HipHop VM 4.36.0-dev (rel)
Compiler: 1576122630_N
Repo schema: aef583187f5c041d84cd3d54b0b066b9409cdde6

$ hh_client --version
hackc-0034fe919032cda5fd9e5ff2ca3a51c290d7fd3e-4.36.0-dev
```

### Operating System and Version

Mac OS Mojave

### Standalone code, or other way to reproduce the problem
```
function example(array<?int, ?int> $example): void {}
```

### Actual result

```
$ hh_client
No errors!
```

### Expected result

HHAST blows up on this code because nullable array keys aren't allowed, I'd expect Hack to complain as well. We discussed whether this should be a bug in HHAST, but I think it's better to ban in Hack. This is what HHAST thinks about thet file:

```
Message: In file "test.hack": Failed to deserialize AST: Argument 3 passed to Facebook\HHAST\MapArrayTypeSpecifier::__construct() must be an instance of Facebook\HHAST\SimpleTypeSpecifier, Facebook\HHAST\NullableTypeSpecifier given
 1  function example(array<?int, ?int> $example): void {}
     ----------------- ^ HERE
```

**Comment by fredemmott** [comment]
FB T59054944

**Comment by lexidor** [comment]
I am going over old issues on this repository, to see which ones apply to the current versions of hhvm.

This code is not accepted in modern hhvm versions. The array type does not exist anymore. It is not possible to sneak this nullability into a dict type. The darray type version if this bug was fixed in [4.11](http://hhvm.com/blog/2019/06/24/hhvm-4.11.0.html). Feel free to reopen the issue if I have missed something.
